core/Pos/Core/Delegation/Util.hs

```diff
@@ -1,28 +0,0 @@
--- | Delegation helpers.
-
-module Pos.Core.Delegation.Util
-       ( checkDlgPayload
-       ) where
-
-import           Universum
-
-import           Control.Monad.Except (MonadError (throwError))
-import           Data.List (groupBy)
-
-import           Pos.Core.Configuration (HasConfiguration)
-import           Pos.Core.Delegation.Types (DlgPayload (..))
-import           Pos.Crypto (ProxySecretKey (..), validateProxySecretKey)
-
--- | Verifier of 'DlgPayload' which ensures absence of duplicates, or invalid
--- PSKs.
-checkDlgPayload :: (HasConfiguration, MonadError Text m) => DlgPayload -> m ()
-checkDlgPayload it = do
-    unless (null duplicates) $
-        throwError "Some of block's PSKs have the same issuer, which is prohibited"
-    forM_ proxySKs validateProxySecretKey
-  where
-    proxySKs = getDlgPayload it
-    proxySKsDups psks =
-        filter (\x -> length x > 1) $
-        groupBy ((==) `on` pskIssuerPk) $ sortOn pskIssuerPk psks
-    duplicates = proxySKsDups proxySKs
```

Comment:
So previously we were checking that there are no such `psk1` and `psk2` that `pskIssuerPk psk1 == pskIssuerPk psk2`, but now it's not checked. Is it ok?

Reply:
Yes. Now `DlgPayload` has a `Set` inside, so the property is guaranteed in the decoder.

Reply:
But `Set` will only guarantee that there are no such `psk1` and `psk2` that `psk1 == psk2`. Isn't it still possible to have `psk1` and `psk2` such that `pskIssuerPk psk1 == pskIssuerPk psk2`?

Reply:
Oh, you're right. No idea how did I miss that. Maybe, I should revert this particular refactoring then.

Reply:
@gromakovsky I've reverted these `list -> set` changes.